lib/core/src/sdk.rs

```diff
@@ -3349,20 +3349,34 @@ impl LiquidSdk {
             None => None,
         };
 
-        let lnurl_pay_domain = match prepare_response.data.ln_address {
-            Some(_) => None,
-            None => Some(prepare_response.data.domain),
-        };
         if let (Some(tx_id), Some(destination)) =
             (payment.tx_id.clone(), payment.destination.clone())
         {
+            // Uses existing `ln_address` if it's available, otherwise fallback to `ln_address` from `metadata_str`
+            let ln_address = derive_ln_address(
+                prepare_response.data.ln_address.clone(),
+                &prepare_response.data.metadata_str,
```

Comment:
You can use `data.metadata_vec()` to parse this
https://github.com/breez/breez-sdk-greenlight/blob/bb68587b221cd1c4a0471b7bc98b3cd0082144f7/libs/sdk-common/src/input_parser.rs#L804-L808